src/actions/csi_benchmark.py

```diff
@@ -0,0 +1,372 @@
+import contextlib
+import dataclasses
+import json
+import logging
+import os
+import shlex
+import shutil
+import subprocess
+import tempfile
+from pathlib import Path
+from typing import Optional
+
+import ops
+from charmhelpers.fetch.archiveurl import ArchiveUrlFetchHandler
+
+log = logging.getLogger(__name__)
+
+BENCH_HOME = Path("/home/ubuntu/kube-bench")
+BENCH_BIN = BENCH_HOME / "kube-bench"
+BENCH_CFG = BENCH_HOME / "cfg-ck"
+GO_PKG = "github.com/aquasecurity/kube-bench"
+RESULTS_DIR = "/home/ubuntu/kube-bench-results"
+
+# Remediation dicts associate a failing test with a Remedy to fix it.
+# Conservative fixes will probably leave the cluster in a good state.
+# Dangerous fixes will likely break the cluster.
+# Tuple examples:
+#  {'1.2.3': Remedy('manual -- we don't know how to auto fix this', None, None)}
+#  {'1.2.3': Remedy('cli', 'command to run', None)}
+#  {'1.2.3': Remedy('kv', 'snap', {cfg_key: value})}
+
+
+@dataclasses.dataclass
+class Remedy:
+    """Remedy class for benchmarking."""
+
+    type: str
+    command: Optional[str]
+    config: Optional[dict] = None
+
+    def run(self, test_num: int, test_remediation: str) -> int:
+        """Run the remedy command."""
+        if self.type == "manual":
+            log.info(
+                "Test %s: unable to auto-apply remedy.\n" "Manual steps:\n%s",
+                test_num,
+                test_remediation,
+            )
+        elif self.type == "cli":
```

Comment:
According to the comment above "kv" is also a possible type. Don't we need to handle this?

Reply:
it doesn't look like the code path deals with that.  see https://github.com/charmed-kubernetes/charm-kubernetes-control-plane/pull/349/files#diff-28032d63f14eb463321b6f903bc8a8ed4f8ee3e76991509f56912ad7157efed7R238-R244